packages/core/src/codewhispererChat/tools/executeBash.ts

```diff
@@ -207,22 +215,68 @@ export class ExecuteBash {
             const stdoutBuffer: string[] = []
             const stderrBuffer: string[] = []
 
-            let firstChunk = true
-            let firstStderrChunk = true
+            // Use a closure boolean value firstChunk and a function to get and set its value atomically
+            let isFirstChunk = true
+            const getAndSetFirstChunk = (newValue: boolean): boolean => {
```

Comment:
the function getAndSetFirstChunk can avoid duplicate code in line 260 and line 271.